packages/web-components/src/search/components/search-result-overlay.ts

```diff
@@ -0,0 +1,127 @@
+import { LitElement, css, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { theme } from '../../theme';
+import { SearchResult } from '../search-overlay';
+
+export class ProductSearchBarResultOverlay extends LitElement {
+
+    @property()
+    setSearchTerm = (term: string) => {};
+
+    @property()
+    noResultsMessage: string | null = null;
+
+    @property({ type: Array })
+    results: SearchResult[] | null = null;
+
+    @property({ type: Number })
+    selectedIndex = -1;
+    
+    connectedCallback(): void {
+        super.connectedCallback();
+    }
+
+    render() {
+        return html`
+            <div class="rw-result-container">
+                ${(!this.results ||
+                this.results.length < 1) ? html`
+                    <div class="rw-no-results">${this.noResultsMessage ?? 'No search results to show'}</div>
+                ` : html`
+                <div>
+                    ${this.results.map((result, index) => {
+                        return html`
+                        <div ?selected=${index === this.selectedIndex} class="rw-selected-result">
+                        ${result.searchTermPrediction ?
+                            html`
+                                <div class="rw-prediction-item-container" @click=${() => this.setSearchTerm(result.searchTermPrediction!.term ?? '')}>
+                                    <h3 class="rw-prediction-item">
+                                        ${result.searchTermPrediction.term}
+                                    </h3>
+                                    <relewise-search-icon class="rw-search-icon"></relewise-search-icon>
+                                </div>
+                            ` : 
+                            html`
+                                <div class="rw-product-item-container">
+                                    <relewise-search-result-overlay-product .product=${result.product ?? null}></relewise-search-result-overlay-product>
```

Comment:
Hvorfor laver du `?? null` her?

Reply:
`export class SearchResult {
    product?: ProductResult;
    searchTermPrediction?: SearchTermPredictionResult;
}`

Da typen på et product her er `ProductResult | undefined`. 🙂

Reply:
Men hvorfor skal du lave det til null her? Kan din component ikke tage imod `undefined`?

Reply:
Bare for at holde consistency med de andre components. 🙂 
Vil du gerne have at jeg retter det?

Reply:
Jeg synes det virker underligt at vi er nød til at lave `undefined` om til `null` i vores interne klasser :)
Hvis kontrakten siger den kan være `undefined`, så tænker jeg da fint at vores component kan tage i mod det.

Reply:
Så er det rettet. 🙂